src/mds/ScrubHeader.h

```diff
@@ -74,6 +92,10 @@ class ScrubHeader {
   bool repaired = false;  // May be set during scrub if repairs happened
   unsigned epoch_last_forwarded = 0;
   unsigned num_pending = 0;
+  // uninline failed info mutex
+  ceph::fair_mutex ufi_mutex;
```

Comment:
I think you can avoid this problem by simply putting this `fair_mutex` in a container (Pimpl C++ pattern). Then including this header won't require knowing its definition because you'll include it in `ScrubHeader.cc`. You'll need to move the definition of these methods to `ScrubHeader.cc` too.

Reply:
I've done away with the fair_mutex; its not required.